pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/DistinctCountHLLPlusPlusMVAggregationFunction.java

```diff
@@ -0,0 +1,225 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.query.aggregation.function;
+
+import com.clearspring.analytics.stream.cardinality.HyperLogLogPlus;
+import java.util.List;
+import java.util.Map;
+import org.apache.pinot.common.request.context.ExpressionContext;
+import org.apache.pinot.core.common.BlockValSet;
+import org.apache.pinot.core.query.aggregation.AggregationResultHolder;
+import org.apache.pinot.core.query.aggregation.groupby.GroupByResultHolder;
+import org.apache.pinot.segment.spi.AggregationFunctionType;
+import org.apache.pinot.spi.data.FieldSpec.DataType;
+
+
+public class DistinctCountHLLPlusPlusMVAggregationFunction extends DistinctCountHLLPlusPlusAggregationFunction {
```

Comment:
Let's use one function to support both SV and MV. You may refer to `DistinctCountThetaSketchAggregationFunction` on how to support both SV and MV in one function

Reply:
Aggregate methods looks bloated with if else for single value and multi value. Shall we leave them as is with overridden functions?

Reply:
I assumed you were talking about code reorganization. If i understood it correctly, expose single aggregate  function for users without having to explicitly specify which one to use based on the schema ?

Reply:
Yes. We can make `DistinctCountHLLPlusPlusAggregationFunction` work on both SV and MV columns. User doesn't need to explicitly use a different function